Makefile

```diff
@@ -44,7 +44,7 @@ build-dev-deps:
 	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b ${GOPATH}/bin v1.10.2
 
 test:
-	go test -race -v -cover -coverprofile=.coverprofile ./...
+	go test -timeout 30s -race -v -cover -coverprofile=.coverprofile ./...
```

Comment:
There shouldn't be a timeout on tests.

Reply:
why not ?

Reply:
There's a default 10min already in place for preventing total test stalls. 30s is much lower, which may start causing flaky behavior when start adding more tests.

Reply:
I'll remove from this PR but I think we should think about adding a smaller timeout than the default.